metadata.json

```diff
@@ -22,7 +22,7 @@
     },
     {
       "name": "puppet-epel",
-      "version_requirement": ">= 3.0.0 < 4.0.0"
```

Comment:
can you please move this commit to a new PR? allowing newer epel isn't related to implementing EL9 support.

Reply:
> can you please move this commit to a new PR? allowing newer epel isn't related to implementing EL9 support.

puppet-epel only supports EL9 sinds [v4.1.0](https://github.com/voxpupuli/puppet-epel/blob/master/CHANGELOG.md#v410-2022-03-26); I feel this PR is senseless without updating this...

Reply:
Also, it (bump epel) is the very title of this PR... I must be too confused by the heat and it being a Friday afternoon.

Reply:
If 4.1.0 is minumum for EPEL9 then the lower bound should be increased also accordingly.

As for if this it should be in this MR at all I think its okay.

Reply:
> If 4.1.0 is minumum for EPEL9 then the lower bound should be increased also accordingly.

I understand the idea, but I'm hesitant here - you are not forced to use EL9, so maybe you want to keep using your current module versions. I know what a hell it is to try and update your modules when they all have some variation of requirements of a shared module (like epel).

Anyway, do you want me to increment the lower bound?

Reply:
this PR contains two changes. Adding EL 9 and allowing the latest EPEL module. Allowing  the latest EPEL module is a requirement for adding EL 9 support. But the change is important on it's own. the collectd module needs to allow the latest EPEL version in case it gets installed into an environment where other modules already require EPEL 4. because of that we should split this PR up into two. one allowing EPEL 4, the other adding code for EL 9.